File: Api/src/FavoDeMel.Domain/Comandas/Comanda.cs
using FavoDeMel.Domain.Common;
using FavoDeMel.Domain.Usuarios;
using System;
using System.Collections.Generic;
using System.Linq;

namespace FavoDeMel.Domain.Comandas
{
    public class Comanda : Entity<int>
    {
        public Usuario Garcom { get; set; }
        public IList<ComandaPedido> Pedidos { get; set; }
        public decimal TotalAPagar { get; set; }
        public decimal GorjetaGarcom { get; set; }
        public ComandaSituacao Situacao { get; set; }

        public Comanda()
        {
            Pedidos = new List<ComandaPedido>();
        }

        public void FecharConta()
        {
            TotalAPagar = Pedidos.Sum(c => c.Quantidade * c.Produto.Preco);
            GorjetaGarcom = (Garcom.Comissao / 100) * TotalAPagar;
            Situacao = ComandaSituacao.Fechada;
        }

        public void Confirmar()
        {
            Situacao = ComandaSituacao.EmAndamento;
        }
    }
}

File: Web/src/app/models/comanda.ts
export class Comanda {
  id: any;
  garcomId: any;
  pedidos: ComandaPedido[];
  totalAPagar: any;
  gorjetaGarcom: any;
  situacao: ComandaSituacao;
}

export class ComandaPedido {
  id: any;
  produtoId: any;
  produtoNome: string;
  quantidade: number;
  situacao: ComandaPedidoSituacao;
}

export enum ComandaSituacao {
  Aberta = 1,
  EmAndamento = 2,
  Fechada = 3
}

export enum ComandaPedidoSituacao {
  Pedido = 1,
  Preparando = 2,
  Pronto = 3
}
